notebooks/features/causal_inference/Effects of Outreach Efforts.ipynb

```diff
@@ -0,0 +1,288 @@
+    "blob_sas_token = \"\"\n",
+    "\n",
+    "# Allow SPARK to read from Blob remotely\n",
+    "wasbs_path = f\"wasbs://{blob_container_name}@{blob_account_name}.blob.core.windows.net/{blob_relative_path}\"\n",
+    "\n",
+    "spark.conf.set(\n",
+    "    f\"fs.azure.sas.{blob_container_name}.{blob_account_name}.blob.core.windows.net\",\n",
+    "    blob_sas_token,\n",
+    ")\n",
+    "\n",
+    "#  Import the sample data\n",
+    "data = spark.read.option(\"inferSchema\", True).option(\"header\", True).csv(wasbs_path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": true,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Data sample\n",
+    "display(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "# Get Causal Effects with SynapseML DoubleMLEstimator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from synapse.ml.causal import *\n",
+    "from pyspark.ml.classification import LogisticRegression\n",
+    "from pyspark.ml.regression import LinearRegression\n",
+    "\n",
+    "treatmentColumn = \"Discount\"\n",
+    "outcomeColumn = \"Revenue\"\n",
+    "\n",
+    "dml = (\n",
+    "    DoubleMLEstimator()\n",
+    "    .setTreatmentModel(LogisticRegression())\n",
+    "    .setTreatmentCol(treatmentColumn)\n",
+    "    .setOutcomeModel(LinearRegression())\n",
+    "    .setOutcomeCol(outcomeColumn)\n",
+    "    .setMaxIter(20)\n",
+    ")\n",
+    "\n",
+    "model = dml.fit(data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Get average treatment effect, it returns a numeric value, e.g. 5166.78324\n",
+    "# It means, on average, customers who received a discount spent $5,166 more on software\n",
+    "model.getAvgTreatmentEffect()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": false,
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Get treatment effect's confidence interval, e.g.  [4765.826181160708, 5371.2817538168965]\n",
```

Comment:
If you really want to make a bang with this demo, consider a quick plot that we can upload to blob and place inline here

Reply:
let me check it with Sarah, what kind of plot we can show here.

Reply:
@mhamilton723 I like the idea of a visualization. Is this along the lines of what you had in mind or were you thinking of something more illustrative of the example?
![ate_ci](https://user-images.githubusercontent.com/22448951/213284486-10412710-6314-4b52-8010-e9f71fce58f6.png)

Reply:
HHhmmm maybe i was too ambitious here it feels silly to do alot of work to just show a single interval. Maybe later we'll take this one up, perhaps for the kind of DoubleML that gets the causal effects of multiple treatments so this interval wont be so lonely lol

Reply:
I also think about it, DoubleML should extend the function to support multiple treatments in the future (It is supported in sparkDML and EconM). With this in mind, I think we should pause the transform function update, and redesign it with multiple treatments, we might want to output the multiple treatments result through transform function. Any thoughts? @mhamilton723 , @sarahshy , @memoryz

Reply:
will re-design this after we add the support for multiple treatments, leave it as is for now.